{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "toc_visible": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "accelerator": "GPU"
  },
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "### This code is **fine tuning DenseNet-161** ( https://github.com/flyyufelix/DenseNet-Keras ) for own dataset. "
      ],
      "metadata": {
        "id": "RwyOKKK3rURT"
      }
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "_qQ62BYEpU60",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "2de1133b-00d2-445d-ccae-d2d0b6e17cb6"
      },
      "source": [
        "!pip install tensorflow-gpu==1.13.1"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: tensorflow-gpu==1.13.1 in /usr/local/lib/python3.7/dist-packages (1.13.1)\n",
            "Requirement already satisfied: astor>=0.6.0 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.13.1) (0.8.1)\n",
            "Requirement already satisfied: gast>=0.2.0 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.13.1) (0.3.3)\n",
            "Requirement already satisfied: keras-preprocessing>=1.0.5 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.13.1) (1.1.2)\n",
            "Requirement already satisfied: keras-applications>=1.0.6 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.13.1) (1.0.8)\n",
            "Requirement already satisfied: tensorflow-estimator<1.14.0rc0,>=1.13.0 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.13.1) (1.13.0)\n",
            "Requirement already satisfied: termcolor>=1.1.0 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.13.1) (1.1.0)\n",
            "Requirement already satisfied: grpcio>=1.8.6 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.13.1) (1.32.0)\n",
            "Requirement already satisfied: wheel>=0.26 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.13.1) (0.36.2)\n",
            "Requirement already satisfied: tensorboard<1.14.0,>=1.13.0 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.13.1) (1.13.1)\n",
            "Requirement already satisfied: numpy>=1.13.3 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.13.1) (1.19.5)\n",
            "Requirement already satisfied: absl-py>=0.1.6 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.13.1) (0.12.0)\n",
            "Requirement already satisfied: six>=1.10.0 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.13.1) (1.15.0)\n",
            "Requirement already satisfied: protobuf>=3.6.1 in /usr/local/lib/python3.7/dist-packages (from tensorflow-gpu==1.13.1) (3.12.4)\n",
            "Requirement already satisfied: h5py in /usr/local/lib/python3.7/dist-packages (from keras-applications>=1.0.6->tensorflow-gpu==1.13.1) (2.10.0)\n",
            "Requirement already satisfied: mock>=2.0.0 in /usr/local/lib/python3.7/dist-packages (from tensorflow-estimator<1.14.0rc0,>=1.13.0->tensorflow-gpu==1.13.1) (4.0.3)\n",
            "Requirement already satisfied: markdown>=2.6.8 in /usr/local/lib/python3.7/dist-packages (from tensorboard<1.14.0,>=1.13.0->tensorflow-gpu==1.13.1) (3.3.4)\n",
            "Requirement already satisfied: werkzeug>=0.11.15 in /usr/local/lib/python3.7/dist-packages (from tensorboard<1.14.0,>=1.13.0->tensorflow-gpu==1.13.1) (1.0.1)\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.7/dist-packages (from protobuf>=3.6.1->tensorflow-gpu==1.13.1) (54.2.0)\n",
            "Requirement already satisfied: importlib-metadata; python_version < \"3.8\" in /usr/local/lib/python3.7/dist-packages (from markdown>=2.6.8->tensorboard<1.14.0,>=1.13.0->tensorflow-gpu==1.13.1) (3.8.1)\n",
            "Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.7/dist-packages (from importlib-metadata; python_version < \"3.8\"->markdown>=2.6.8->tensorboard<1.14.0,>=1.13.0->tensorflow-gpu==1.13.1) (3.4.1)\n",
            "Requirement already satisfied: typing-extensions>=3.6.4; python_version < \"3.8\" in /usr/local/lib/python3.7/dist-packages (from importlib-metadata; python_version < \"3.8\"->markdown>=2.6.8->tensorboard<1.14.0,>=1.13.0->tensorflow-gpu==1.13.1) (3.7.4.3)\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "nfnLgiyvpqgi",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "2f9e2b06-a7a3-4ee9-86fe-20de2ee96ee5"
      },
      "source": [
        "!pip install keras==2.0.5"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: keras==2.0.5 in /usr/local/lib/python3.7/dist-packages (2.0.5)\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.7/dist-packages (from keras==2.0.5) (1.15.0)\n",
            "Requirement already satisfied: pyyaml in /usr/local/lib/python3.7/dist-packages (from keras==2.0.5) (3.13)\n",
            "Requirement already satisfied: theano in /usr/local/lib/python3.7/dist-packages (from keras==2.0.5) (1.0.5)\n",
            "Requirement already satisfied: scipy>=0.14 in /usr/local/lib/python3.7/dist-packages (from theano->keras==2.0.5) (1.4.1)\n",
            "Requirement already satisfied: numpy>=1.9.1 in /usr/local/lib/python3.7/dist-packages (from theano->keras==2.0.5) (1.19.5)\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "cU7tzZVXvUuY"
      },
      "source": [
        "import matplotlib.pyplot as plt"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "4KofJKJy9svO",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "ce74a874-b896-4f2e-ba53-a7ffbe6822c0"
      },
      "source": [
        "cd /content"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/content\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Loading dataset, Model, and code"
      ],
      "metadata": {
        "id": "RT73zCJptFI1"
      }
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "aU8zzqtc92RN",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "f76f2d36-e29c-4b44-e2bf-80584d3380fc"
      },
      "source": [
        "import zipfile\n",
        "from google.colab import drive\n",
        "\n",
        "drive.mount('/content/drive/')\n",
        "\n",
        "zip_ref = zipfile.ZipFile(\"/content/drive/My Drive/DenseNet_Civil_complete.zip\", 'r')\n",
        "zip_ref.extractall(\"\")\n",
        "zip_ref.close()"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Drive already mounted at /content/drive/; to attempt to forcibly remount, call drive.mount(\"/content/drive/\", force_remount=True).\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5iC1oXqjt43x",
        "outputId": "8bc9b5dd-b197-4acc-d465-17210adc73c7"
      },
      "source": [
        "cd /content/DenseNet_Civil_complete"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/content/DenseNet_Civil_complete\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "zMuBxZmWv5vY",
        "outputId": "eb7862f7-8b63-46ec-c96f-2a3552d81974"
      },
      "source": [
        "!python densenet161-civil.py"
      ],
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Using TensorFlow backend.\n",
            "/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/dtypes.py:516: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
            "  _np_qint8 = np.dtype([(\"qint8\", np.int8, 1)])\n",
            "/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/dtypes.py:517: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
            "  _np_quint8 = np.dtype([(\"quint8\", np.uint8, 1)])\n",
            "/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/dtypes.py:518: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
            "  _np_qint16 = np.dtype([(\"qint16\", np.int16, 1)])\n",
            "/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/dtypes.py:519: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
            "  _np_quint16 = np.dtype([(\"quint16\", np.uint16, 1)])\n",
            "/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/dtypes.py:520: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
            "  _np_qint32 = np.dtype([(\"qint32\", np.int32, 1)])\n",
            "/usr/local/lib/python3.7/dist-packages/tensorflow/python/framework/dtypes.py:525: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
            "  np_resource = np.dtype([(\"resource\", np.ubyte, 1)])\n",
            "/usr/local/lib/python3.7/dist-packages/tensorboard/compat/tensorflow_stub/dtypes.py:541: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
            "  _np_qint8 = np.dtype([(\"qint8\", np.int8, 1)])\n",
            "/usr/local/lib/python3.7/dist-packages/tensorboard/compat/tensorflow_stub/dtypes.py:542: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
            "  _np_quint8 = np.dtype([(\"quint8\", np.uint8, 1)])\n",
            "/usr/local/lib/python3.7/dist-packages/tensorboard/compat/tensorflow_stub/dtypes.py:543: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
            "  _np_qint16 = np.dtype([(\"qint16\", np.int16, 1)])\n",
            "/usr/local/lib/python3.7/dist-packages/tensorboard/compat/tensorflow_stub/dtypes.py:544: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
            "  _np_quint16 = np.dtype([(\"quint16\", np.uint16, 1)])\n",
            "/usr/local/lib/python3.7/dist-packages/tensorboard/compat/tensorflow_stub/dtypes.py:545: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
            "  _np_qint32 = np.dtype([(\"qint32\", np.int32, 1)])\n",
            "/usr/local/lib/python3.7/dist-packages/tensorboard/compat/tensorflow_stub/dtypes.py:550: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
            "  np_resource = np.dtype([(\"resource\", np.ubyte, 1)])\n",
            "WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:431: The name tf.placeholder is deprecated. Please use tf.compat.v1.placeholder instead.\n",
            "\n",
            "densenet161-civil.py:65: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(96, (7, 7), name=\"conv1\", strides=(2, 2), use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 7, 7, subsample=(2, 2), name='conv1', bias=False)(x)\n",
            "WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:3445: The name tf.random_uniform is deprecated. Please use tf.random.uniform instead.\n",
            "\n",
            "WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:112: The name tf.get_default_graph is deprecated. Please use tf.compat.v1.get_default_graph instead.\n",
            "\n",
            "WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:3288: The name tf.nn.max_pool is deprecated. Please use tf.nn.max_pool2d instead.\n",
            "\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv2_1_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv2_1_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:205: UserWarning: The `merge` function is deprecated and will be removed after 08/2017. Use instead layers from `keras.layers.merge`, e.g. `add`, `concatenate`, etc.\n",
            "  concat_feat = merge([concat_feat, x], mode='concat', concat_axis=concat_axis, name='concat_'+str(stage)+'_'+str(branch))\n",
            "/usr/local/lib/python3.7/dist-packages/keras/legacy/layers.py:460: UserWarning: The `Merge` layer is deprecated and will be removed after 08/2017. Use instead layers from `keras.layers.merge`, e.g. `add`, `concatenate`, etc.\n",
            "  name=name)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv2_2_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv2_2_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv2_3_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv2_3_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv2_4_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv2_4_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv2_5_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv2_5_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv2_6_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv2_6_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:176: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv2_blk\", use_bias=False)`\n",
            "  x = Convolution2D(int(nb_filter * compression), 1, 1, name=conv_name_base, bias=False)(x)\n",
            "WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:3290: The name tf.nn.avg_pool is deprecated. Please use tf.nn.avg_pool2d instead.\n",
            "\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv3_1_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv3_1_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv3_2_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv3_2_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv3_3_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv3_3_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv3_4_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv3_4_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv3_5_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv3_5_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv3_6_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv3_6_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv3_7_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv3_7_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv3_8_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv3_8_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv3_9_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv3_9_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv3_10_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv3_10_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv3_11_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv3_11_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv3_12_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv3_12_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:176: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(384, (1, 1), name=\"conv3_blk\", use_bias=False)`\n",
            "  x = Convolution2D(int(nb_filter * compression), 1, 1, name=conv_name_base, bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_1_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_1_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_2_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_2_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_3_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_3_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_4_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_4_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_5_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_5_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_6_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_6_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_7_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_7_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_8_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_8_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_9_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_9_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_10_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_10_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_11_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_11_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_12_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_12_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_13_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_13_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_14_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_14_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_15_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_15_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_16_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_16_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_17_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_17_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_18_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_18_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_19_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_19_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_20_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_20_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_21_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_21_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_22_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_22_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_23_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_23_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_24_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_24_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_25_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_25_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_26_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_26_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_27_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_27_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_28_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_28_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_29_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_29_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_30_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_30_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_31_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_31_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_32_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_32_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_33_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_33_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_34_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_34_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_35_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_35_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv4_36_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv4_36_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:176: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(1056, (1, 1), name=\"conv4_blk\", use_bias=False)`\n",
            "  x = Convolution2D(int(nb_filter * compression), 1, 1, name=conv_name_base, bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_1_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_1_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_2_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_2_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_3_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_3_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_4_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_4_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_5_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_5_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_6_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_6_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_7_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_7_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_8_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_8_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_9_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_9_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_10_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_10_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_11_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_11_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_12_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_12_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_13_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_13_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_14_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_14_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_15_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_15_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_16_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_16_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_17_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_17_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_18_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_18_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_19_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_19_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_20_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_20_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_21_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_21_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_22_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_22_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_23_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_23_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "densenet161-civil.py:139: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(192, (1, 1), name=\"conv5_24_x1\", use_bias=False)`\n",
            "  x = Convolution2D(inter_channel, 1, 1, name=conv_name_base+'_x1', bias=False)(x)\n",
            "densenet161-civil.py:149: UserWarning: Update your `Conv2D` call to the Keras 2 API: `Conv2D(48, (3, 3), name=\"conv5_24_x2\", use_bias=False)`\n",
            "  x = Convolution2D(nb_filter, 3, 3, name=conv_name_base+'_x2', bias=False)(x)\n",
            "WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:153: The name tf.get_default_session is deprecated. Please use tf.compat.v1.get_default_session instead.\n",
            "\n",
            "WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:158: The name tf.ConfigProto is deprecated. Please use tf.compat.v1.ConfigProto instead.\n",
            "\n",
            "2021-04-11 16:44:46.650851: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA\n",
            "2021-04-11 16:44:46.655803: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2199995000 Hz\n",
            "2021-04-11 16:44:46.656022: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55a8ca9e1880 executing computations on platform Host. Devices:\n",
            "2021-04-11 16:44:46.656056: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>\n",
            "2021-04-11 16:44:46.657804: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcuda.so.1\n",
            "2021-04-11 16:44:46.795701: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2021-04-11 16:44:46.796459: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55a8ca9e1c00 executing computations on platform CUDA. Devices:\n",
            "2021-04-11 16:44:46.796494: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): Tesla T4, Compute Capability 7.5\n",
            "2021-04-11 16:44:46.796673: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2021-04-11 16:44:46.797227: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
            "name: Tesla T4 major: 7 minor: 5 memoryClockRate(GHz): 1.59\n",
            "pciBusID: 0000:00:04.0\n",
            "2021-04-11 16:44:46.797581: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
            "2021-04-11 16:44:46.798897: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
            "2021-04-11 16:44:46.800064: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcufft.so.10.0\n",
            "2021-04-11 16:44:46.800487: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcurand.so.10.0\n",
            "2021-04-11 16:44:46.801778: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusolver.so.10.0\n",
            "2021-04-11 16:44:46.802749: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcusparse.so.10.0\n",
            "2021-04-11 16:44:46.805841: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
            "2021-04-11 16:44:46.805997: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2021-04-11 16:44:46.806661: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2021-04-11 16:44:46.807206: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1763] Adding visible gpu devices: 0\n",
            "2021-04-11 16:44:46.807271: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudart.so.10.0\n",
            "2021-04-11 16:44:46.808499: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
            "2021-04-11 16:44:46.808524: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      0 \n",
            "2021-04-11 16:44:46.808536: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1200] 0:   N \n",
            "2021-04-11 16:44:46.808655: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2021-04-11 16:44:46.809235: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:1005] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n",
            "2021-04-11 16:44:46.809886: W tensorflow/core/common_runtime/gpu/gpu_bfc_allocator.cc:40] Overriding allow_growth setting because the TF_FORCE_GPU_ALLOW_GROWTH environment variable is set. Original config value was 0.\n",
            "2021-04-11 16:44:46.809941: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1326] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 14257 MB memory) -> physical GPU (device: 0, name: Tesla T4, pci bus id: 0000:00:04.0, compute capability: 7.5)\n",
            "WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/optimizers.py:699: The name tf.train.Optimizer is deprecated. Please use tf.compat.v1.train.Optimizer instead.\n",
            "\n",
            "densenet161-civil.py:251: UserWarning: The `nb_epoch` argument in `fit` has been renamed `epochs`.\n",
            "  validation_data=(X_valid1, Y_valid1),\n",
            "WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/tensorflow/python/ops/math_grad.py:1250: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.\n",
            "Instructions for updating:\n",
            "Use tf.where in 2.0, which has the same broadcast rule as np.where\n",
            "WARNING:tensorflow:Variable *= will be deprecated. Use `var.assign(var * other)` if you want assignment to the variable value or `x = x * y` if you want a new python Tensor object.\n",
            "WARNING:tensorflow:From /usr/local/lib/python3.7/dist-packages/keras/backend/tensorflow_backend.py:601: calling Constant.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
            "Instructions for updating:\n",
            "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
            "Train on 180 samples, validate on 94 samples\n",
            "Epoch 1/10\n",
            "2021-04-11 16:45:50.271197: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0\n",
            "2021-04-11 16:45:50.608299: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
            "180/180 [==============================] - 59s - loss: 0.8608 - acc: 0.5167 - val_loss: 0.8712 - val_acc: 0.4894\n",
            "Epoch 2/10\n",
            "180/180 [==============================] - 13s - loss: 0.7927 - acc: 0.8056 - val_loss: 0.8283 - val_acc: 0.7340\n",
            "Epoch 3/10\n",
            "180/180 [==============================] - 13s - loss: 0.7526 - acc: 0.8667 - val_loss: 0.7853 - val_acc: 0.9043\n",
            "Epoch 4/10\n",
            "180/180 [==============================] - 13s - loss: 0.7197 - acc: 0.9056 - val_loss: 0.7752 - val_acc: 0.8511\n",
            "Epoch 5/10\n",
            "180/180 [==============================] - 13s - loss: 0.7143 - acc: 0.9333 - val_loss: 0.7543 - val_acc: 0.9149\n",
            "Epoch 6/10\n",
            "180/180 [==============================] - 13s - loss: 0.7017 - acc: 0.9556 - val_loss: 0.7272 - val_acc: 0.9681\n",
            "Epoch 7/10\n",
            "180/180 [==============================] - 13s - loss: 0.7048 - acc: 0.9500 - val_loss: 0.7123 - val_acc: 0.9681\n",
            "Epoch 8/10\n",
            "180/180 [==============================] - 13s - loss: 0.6955 - acc: 0.9722 - val_loss: 0.7250 - val_acc: 0.9468\n",
            "Epoch 9/10\n",
            "180/180 [==============================] - 13s - loss: 0.6890 - acc: 0.9778 - val_loss: 0.7096 - val_acc: 0.9468\n",
            "Epoch 10/10\n",
            "180/180 [==============================] - 13s - loss: 0.6925 - acc: 0.9611 - val_loss: 0.6924 - val_acc: 0.9787\n",
            "94/94 [==============================] - 6s     \n",
            "94/94 [==============================] - 1s     \n",
            "Prediction accuracy: 1.0\n"
          ],
          "name": "stdout"
        }
      ]
    }
  ]
}